src/lib/testdroid.js

```diff
@@ -0,0 +1,455 @@
+import { version } from '../../package.json';
+import Debug from 'debug';
+import request from 'superagent-promise';
+import urljoin from 'url-join';
+import util from 'util';
+import Project from './project';
+
+let debug = Debug('testdroid-client');
+
+function sleep(duration) {
+  return new Promise(function(accept) {
+    setTimeout(accept, duration);
+    });
+}
+
+/**
+ * Initializes a new Testdroid client to be used with the Testdroid Cloud API.
+ *
+ * Examples:
+ *
+ * var Testdroid = require('testdroid-client');
+
+ * var username = 'joe@example.com';
+ * var password = '123456';
+ * var cloudUrl = 'http://cloudurl/';
+
+ * var client = new Testdroid(cloudUrl, username, password);
+
+ * // Get list of devices
+
+ * client.getDevices().then(function(devices) {
+    console.dir(getDevices);
+ * });
+ *
+ * @param {String} url - URL of the cloud api
+ * @param {String} username
+ * @param {String} password
+ */
+export default class {
+  constructor(url, username, password) {
+    this.version = version;
+    this.baseUrl = url;
+    this.apiUrl = url + 'api/v2/';
+    this.username = username;
+    this.password = password;
+    this.userAgent = `testdroid-client/${this.version}`;
+  }
+
+  /**
+   * Submits a request to the api endpoint.  Options can be passed in for payload and
+   * additional headers.
+   *
+   * Example options:
+   * var requestOptions = {
+   *   'headers': {
+   *     'x-user': 'foo'
+   *    },
+   *    'payload': {
+   *      'limit': 1
+   *    }
+   * };
+   *
+   * __request('get', '/devices', requestOptions);
+   *
+   * @param {String} method - method for the request.  'get' or 'post'
+   * @param {String} path - endpoint to submit request to
+   * @param {opts} opts - optional opts to include.  Maybe include payload and/or headers
+   * @returns {Object} Response
+   */
+  async __request(method, path, opts) {
```

Comment:
This is a longer comment about the design here... I tend to put what I consider "private" (or more specifically this code is just helpers that have nothing to do with this object) as stand alone functions so it is clear that nobody else should call them and so it is clear they don't depend on the state of this class/object/thing